daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs

```diff
@@ -267,6 +265,45 @@ runStart
             case mbJsonApiPort of
                 Nothing -> f sandboxPh
                 Just jsonApiPort -> withJsonApi sandboxPort jsonApiPort args f
+        doCodegen doClean projectConfig =
+          forM_ [minBound :: Lang .. maxBound :: Lang] $ \lang -> do
+            mbOutputPath :: Maybe String <-
+              requiredE ("Failed to parse codegen entry for " <> showLang lang) $
+              queryProjectConfig
+                ["codegen", showLang lang, "output-directory"]
+                projectConfig
+            whenJust mbOutputPath $ \outputPath -> do
+              when doClean $ removeDirectoryRecursive outputPath
+              runCodegen lang []
+        doUploadDar =
+          runLedgerUploadDar (LedgerFlags Nothing Nothing Nothing Nothing 10) Nothing
+        listenForKeyPress projectConfig = do
+          hSetBuffering stdin NoBuffering
+          void $
+            forkIO $
+             do
+              threadDelay 20000000
+              -- give sandbox 20 secs to startup before printing rebuild instructions
+              forever $ do
+                printRebuildInstructions
+                c <- getChar
+                when (c == 'r') $ rebuild projectConfig
```

Comment:
Might want to check for `'R'` as well ... who knows, they might have caps lock on.

```suggestion
                when (c `elem` "rR") $ rebuild projectConfig
```

Reply:
I like it! Thanks!